src/distdl/nn/padnd.py

```diff
@@ -1,46 +1,23 @@
 import numpy as np
 import torch
-
-
-class PadNdFunction(torch.autograd.Function):
-
-    @staticmethod
-    def forward(ctx, input, pad_width, value):
-
-        ctx.pad_width = pad_width
-
-        input_numpy = input.detach().numpy()
-
-        result = np.pad(input_numpy, pad_width, mode='constant', constant_values=value)
-
-        return torch.tensor(result, requires_grad=input.requires_grad)
-
-    @staticmethod
-    def backward(ctx, grad_output):
-
-        pad_width = ctx.pad_width
-
-        slices = []
-        for (lpad, rpad) in pad_width:
-            start = lpad
-            stop = -rpad if rpad > 0 else None
-            slices.append(slice(start, stop, 1))
-
-        grad_output_numpy = grad_output.detach().numpy()
-
-        result = grad_output_numpy[tuple(slices)]
-
-        return torch.tensor(result, requires_grad=grad_output.requires_grad), None, None, None
+import torch.nn.functional as F
 
 
 class PadNd(torch.nn.Module):
 
     def __init__(self, pad_width, value):
 
         super(PadNd, self).__init__()
-
         self.pad_width = pad_width
         self.value = value
+        self.torch_pad = self._to_torch_padding(self.pad_width)
+
+    def _to_torch_padding(self, pad):
```

Comment:
Is this still used?  I thought we removed it.  Should we just remove the layer itself?

Reply:
I don't think the layer is being used anywhere, and since it's just a local padding, the only use is for interoperability between how we represent padding, to how PyTorch represents padding.

Somewhat related, I think what would be useful to users down the road is a distributed pad. That's something that can be used in other layers too.

Reply:
If you take this out, maybe we just remove it in another PR